### PR TITLE
1617 flicker on space bar

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts [hunk 1]
```diff
@@ -110,6 +163,31 @@ export class CoreAppEffects {
...
+	@Effect()
+	onNextPresetOverlay$: Observable<any> = this.actions$
+		.ofType<GoNextPresetOverlay>(CoreActionTypes.GO_NEXT_PRESET_OVERLAY)
+		.withLatestFrom(this.store$.select(mapStateSelector), (Action , mapState: IMapState): {overlayId: string, mapId: string} => {
+			const activeMap = MapFacadeService.activeMap(mapState);
+			return {overlayId: activeMap.data.overlay && activeMap.data.overlay.id, mapId: mapState.activeMapId};
+		})
+		.withLatestFrom(this.store$.select(coreStateSelector), ({overlayId , mapId}, { presetOverlays }): {overlay: IOverlay, mapId: string} => {
+			const length = presetOverlays.length;
+			if (length === 0) { return }
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts]: return;

File: src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts [hunk 2]
```diff
@@ -110,6 +163,31 @@ export class CoreAppEffects {
...
+			const index = presetOverlays.findIndex(overlay => overlay.id === overlayId),
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts]: const index = presetOverlays.findIndex(overlay => overlay.id === overlayId);
const ...

File: src/app/@ansyn/core/components/imagery-status/imagery-status.component.ts [hunk 3]
```diff
@@ -134,6 +154,14 @@ export class ImageryStatusComponent implements OnInit, OnDestroy {
 		this.favoritesButtonText = this.isFavorite ? 'Remove from favorites' : 'Add to favorites';
 	}
 
+	updatePresetStatus() {
+		this.isPreset = false;
+		if (this.overlay && this.presetOverlays && this.presetOverlays.length > 0) {
+			this.isPreset = this.presetOverlays.some(o => o.id === this.overlay.id);
+		}
+		this.presetsButtonText = this.isPreset ? 'Remove from presets' : 'Add to presets (press F to traverse)';
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/core/components/imagery-status/imagery-status.component.ts]: done

File: src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts]: done

File: src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts (hunk 2 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/core.app.effects.ts]: done

**Pull request merged**